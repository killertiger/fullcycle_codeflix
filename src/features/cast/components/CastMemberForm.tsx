import { Box, Button, FormControl, FormControlLabel, FormGroup, FormLabel, Grid, Radio, RadioGroup, TextField } from "@mui/material";
import { Link } from "react-router-dom";
import { CastMember } from "../../../types/CastMembers";

type Props = {
    castMember: CastMember;
    isDisabled?: boolean;
    isLoading?: boolean;
    handleSubmit: (e: React.FormEvent<HTMLFormElement>) => void;
    handleChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
}

export function CastMemberForm({
    castMember,
    isDisabled = false,
    isLoading = false,
    handleSubmit,
    handleChange,
}: Props) {
    return (
        <Box p={2}>
            <form onSubmit={handleSubmit}>
                <Grid container spacing={3}>
                    <Grid item xs={12}>
                        <FormControl fullWidth>
                            <TextField
                                required
                                name="name"
                                label="name"
                                value={castMember.name}
                                disabled={isDisabled}
                                onChange={handleChange}
                            />
                        </FormControl>
                    </Grid>

                    <Grid item xs={12}>
                        <FormGroup>
                            <FormLabel>Type</FormLabel>
                        </FormGroup>
                        <RadioGroup
                            aria-labelledby="type of cast member"
                            defaultValue="Director"
                            name="type"
                            onChange={handleChange}
                            value={castMember.type}>
                            <FormControlLabel value={1} control={<Radio />} label="Director" />
                            <FormControlLabel value={2} control={<Radio />} label="Actor" />
                        </RadioGroup>
                    </Grid>

                    <Grid item xs={12}>
                        <Box display="flex" gap={2}>
                            <Button variant="contained" component={Link} to="/cast-members">
                                Back
                            </Button>

                            <Button type="submit" variant="contained" color="secondary" disabled={isDisabled || isLoading}>
                                {isLoading ? "Loading..." : "Save"}
                            </Button>
                        </Box>
                    </Grid>
                </Grid>
            </form>
        </Box>
    )
}