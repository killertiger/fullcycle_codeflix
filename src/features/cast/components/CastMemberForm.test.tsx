import { render } from '@testing-library/react';
import { CastMemberForm } from './CastMemberForm';
import { BrowserRouter } from 'react-router-dom';


const Props = {
    castMember: {
        id: "1",
        name: "test",
        type: 1,
        deleted_at: null,
        created_at: "2023-12-13T04:17:44.000000Z",
        updated_at: "2023-12-13T04:17:44.000000Z"
    },
    isDisabled: false,
    isLoading: false,
    handleSubmit: jest.fn(),
    handleChange: jest.fn(),
}

describe("CastMemberForm", () => {
    it("should render castMember form correctly", () => {
        const { asFragment } = render(<CastMemberForm {...Props} />,
        {
            wrapper: BrowserRouter
        })

        expect(asFragment).toMatchSnapshot();
    })
})