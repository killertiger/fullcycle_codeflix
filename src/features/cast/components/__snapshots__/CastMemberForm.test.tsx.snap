// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CastMemberForm should render castMember form correctly 1`] = `[Function]`;
